### DISCUSSION TITLE
Does micronaut have existing api gateway library?

### DISCUSSION THREAD

**Discussion by EthanYe0630** [micronaut-projects/micronaut-core#7218] [Q&A]
Hi all,

I'm trying to add api gateway feature before my micronaut service. I couldn't find any related posts. Is it available?

I only found a solution for spring https://springframework.guru/api-gateway/#:~:text=The%20API%20Gateway%20Service%20is,requests%20to%20the%20Message%20service.&text=main%20class%20to%20enable%20Zuul

**Comment by sdelamo** [comment]
No, we currently don' thave an Gateway solution.

**Reply by alexanderankin** [reply]
@sdelamo - If I was interested in creating one, what would be the most beneficial use of my time to users of the micronaut framework: should I go off on my own and try to implement some features of spring's gateway project or is it realistic for a hobby-level micronaut practitioner like myself to actually craft a micronaut guide and a project that you guys might accept into your maintenance? We might need to use a gateway on a project, and I would love to push for a micronaut solution but the ability to configure endpoints from yaml out of the box was a huge selling point to my team (a feature of the spring gateway toolbox). Thanks in advance!

**Comment by yawkat** [comment]
I'm a bit wary of officially supporting such a module. It is very security-sensitive and would need careful attention. There is *lots* of potential for security bugs.
